internal/cmd/controller/cleanup/root.go

```diff
@@ -0,0 +1,33 @@
+package cleanup
+
+import (
+	"fmt"
+
+	command "github.com/rancher/fleet/internal/cmd"
+	"github.com/rancher/fleet/pkg/version"
+	"github.com/spf13/cobra"
+)
+
+type CleanUp struct {
+	Kubeconfig string `usage:"kubeconfig file"`
+	Namespace  string `usage:"namespace to watch" env:"NAMESPACE"`
+}
+
+func (c *CleanUp) Run(cmd *cobra.Command, args []string) error {
+	if c.Namespace == "" {
+		return fmt.Errorf("--namespace or env NAMESPACE is required to be set")
+	}
+	if err := start(cmd.Context(), c.Kubeconfig, c.Namespace); err != nil {
+		return err
+	}
+	<-cmd.Context().Done()
+
+	return nil
```

Comment:
Why is this needed? `RunOrDie` used in `start` is blocking, isn't it?
https://github.com/rancher/wrangler/blob/v2.1.2/pkg/leader/leader.go#L29

```suggestion
	return start(cmd.Context(), c.Kubeconfig, c.Namespace)
```

Reply:
I must have added it because I copied pasted from the [agent](https://github.com/rancher/fleet/blob/master/internal/cmd/agent/root.go#L48)